### ISSUE TITLE
fst and gson's LinkedTreeMap/LinkedHashTreeMap

### ISSUE THREAD

**Issue by dguggi** [RuedigerMoeller/fast-serialization#66]
the following test serializes/deserializes a LinkedTreeMap (fst v2.24/2.25/2.27):

```java
    @Test
    public void serialize() throws Exception {
        FSTConfiguration fst = FSTConfiguration.createDefaultConfiguration();
        OutputStream out = new FileOutputStream("/home/dguggi/test");
        ObjectOutput output = fst.getObjectOutput(out);
        LinkedTreeMap map = new LinkedTreeMap();
        map.put("key", "value");
        output.writeObject(map);
        output.flush();

        InputStream in = new FileInputStream("/home/dguggi/test");
        System.out.println(fst.getObjectInput(in).readObject());
    }
```
which leads to the following exception:

java.io.IOException: java.lang.NullPointerException
	at org.nustaq.serialization.FSTObjectInput.readObject(FSTObjectInput.java:241)
	...

By adding the following lines serialization/deserialization works:
```java
fst.registerSerializer(LinkedTreeMap.class, new FSTMapSerializer(), true);
fst.registerSerializer(LinkedHashTreeMap.class, new FSTMapSerializer(), true);
```

**Comment by RuedigerMoeller** [comment]
Thanks for reporting ! Actually there are some rare edge cases where classes rely on implementation specific behaviour of stock JDK serialization which are impossible to resolve for an alternative implementation like fst. Seems like a pattern these are exploited by google#s code (android) ;) .
I'll have a look into that, though the outcome probably is to workaround like you proposed.

**Comment by RuedigerMoeller** [comment]
Added those two to Android default initialization. Thanks for reporting.
Released with 2.30
